server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -1967,7 +1967,10 @@ public void deletePools(Collection<Pool> pools, Set<String> alreadyDeletedPools)
             }
 
             poolsToDelete.add(pool);
-            entitlementsToRevoke.addAll(pool.getEntitlements());
+            for (Entitlement e: pool.getEntitlements()) {
+                e.setDeletedFromPool(true);
+                entitlementsToRevoke.add(e);
+            }
```

Comment:
This is a good change, and one that I've made myself (in a sense) in the deletePools refactor; but it's one that may have fallout. It might be worth investigating whether or not people will be expecting those events in this case. If they are, we're about to break them -- despite it being incorrect behavior.

Reply:
W.r.t. this issue, fwiw , I share @Ceiu 's concern in this case, and here are some facts about it.
* there are two implementations which delete pool and their entitlements, and in master today, in deletePool we send out events for the entitlement deletes as well as the pool deletes, but in deletePools we send out events only for the pool deletes, and considers the entitlement deletes redundant.
* Since this was an intentional implementation in deletePools, This PR makes candlepin consistent in this aspect, by skipping those events in deletePool.

Reply:
Based on conversation with @barnabycourt  , This change is okay . @barnabycourt  please confirm.